modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseUpgradeCheck.java

```diff
@@ -46,13 +54,85 @@ protected String addNewImports(String newContent) {
 		return newContent;
 	}
 
+	protected String addNewImportsJSP(String newContent, String... newImports) {
+		if (newImports.length == 0) {
+			return newContent;
+		}
+
+		Set<String> missingImports = new TreeSet<>();
+
+		Collections.addAll(missingImports, newImports);
+
+		if (missingImports.isEmpty()) {
+			return newContent;
+		}
+
+		Matcher directiveMatcher = _directivePattern.matcher(newContent);
+
+		if (directiveMatcher.find()) {
+			String includeDirective = directiveMatcher.group();
+
+			String[] includeDirectives = StringUtil.splitLines(
+				includeDirective);
+
+			String newHeaderPackage = createImportsPackageJsp(
+				missingImports, includeDirectives);
+
+			newContent = StringUtil.replaceFirst(
+				newContent,
+				com.liferay.petra.string.StringUtil.merge(
+					includeDirectives, "\n"),
+				newHeaderPackage);
+		}
+		else {
+			Matcher commentMatcher = _commentsPattern.matcher(newContent);
+
+			if (commentMatcher.find()) {
+				String comment = commentMatcher.group(3);
```

Comment:
@TamyrisTorres, how do you ensure that the comments are in the beginning of the content?

Reply:
resolved.